.github/actions/setup-binary/action.yml

```diff
@@ -158,10 +158,11 @@ runs:
       shell: 'bash'
       env:
         INSTALL_PATH: '${{ inputs.install_path }}'
+        BINARY_SUBPATH: '${{ inputs.binary_subpath }}'
       run: |-
         # Add binary to path
-        echo "Adding ${INSTALL_PATH} to PATH"
-        echo "${INSTALL_PATH}" >> $GITHUB_PATH
+        echo "Adding ${INSTALL_PATH}/${BINARY_SUBPATH} to PATH"
```

Comment:
Do we need to consider destination_subpath at all? We might want to move this to the end consider which to add to path.

Reply:
That's a good point, let me think how to re-organize the steps.

Reply:
Ok, I think this should work:
1. MV install_path / binary_subpath -> install_path / destination_subpath
2. If destination_subpath is present, binary_subpath = destination_subpath.
    Add install_path / binary_subpath to PATH.
4. If destination_subpath is present, binary_subpath = destination_subpath.
    Mark install_path / binary_subpath as executable.